plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-server/src/main/java/org/eclipse/che/plugin/java/plain/server/projecttype/PlainJavaInitHandler.java

```diff
@@ -26,12 +35,41 @@
  * @author Valeriy Svydenko
  */
 public class PlainJavaInitHandler implements ProjectInitHandler {
+  private final Provider<JavaLanguageServerExtensionService> extensionService;
+  private final Provider<JavaLanguageServerLauncher> languageServerLauncher;
+  private final Provider<ProjectManager> projectRegistryProvider;
+
+  @Inject
+  public PlainJavaInitHandler(
+      Provider<JavaLanguageServerExtensionService> extensionService,
+      Provider<JavaLanguageServerLauncher> languageServerLauncher,
+      Provider<ProjectManager> projectRegistryProvider) {
+    this.extensionService = extensionService;
+    this.languageServerLauncher = languageServerLauncher;
+    this.projectRegistryProvider = projectRegistryProvider;
+  }
+
   @Override
   public String getProjectType() {
     return JAVAC;
   }
 
   @Override
   public void onProjectInitialized(String projectFolder)
-      throws ServerException, ForbiddenException, ConflictException, NotFoundException {}
+      throws ServerException, ForbiddenException, ConflictException, NotFoundException {
+    if (!languageServerLauncher.get().isStarted()) {
+      return;
+    }
+    String wsPath = absolutize(projectFolder);
+    ProjectConfig project =
+        projectRegistryProvider
+            .get()
+            .get(wsPath)
+            .orElseThrow(() -> new ServerException("Can't find a project: " + wsPath));
+
+    List<String> library = project.getAttributes().get(LIBRARY_FOLDER);
```

Comment:
If this property is set, shouldn't the classpath have been updated before? I'm not sure why we need this code.

Reply:
This code executes when the property was set and if this property exists we need to update classpath to add jars from this folder.